network/transport/v2/protocol.go

```diff
@@ -240,6 +240,9 @@ func (p *protocol) handlePrivateTxRetry(event dag.Event) (bool, error) {
 	// Sanity check: if we have the payload, mark this job as finished
 	payload, err := p.state.ReadPayload(context.Background(), event.Transaction.PayloadHash())
 	if err != nil {
+		if !errors.Is(err, stoabs.ErrDatabase{}) {
```

Comment:
Misuse of errors.Is here. Use `errors.As(err, new(stoabs.ErrDatabase))` instead.

Reply:
I'm proposing to let `stoabs.DatabaseError` return `stoabs.ErrDatabase` so we can use `errors.As(err, new(stoabs.ErrDatabase))`, see https://github.com/nuts-foundation/go-stoabs/pull/29

Reply:
The pattern is good. Just the type name s/stoabs.ErrDatabase/stoabs.DatabaseError/ again.

Reply:
different library / larger discussion